packages/lib-classifier/src/store/WorkflowStepStore.js

```diff
@@ -79,13 +72,41 @@ const WorkflowStepStore = types
       addDisposer(self, workflowDisposer)
     }
 
+    function createAnnotationObserver () {
+      const annotationDisposer = autorun(() => {
+        if (self.active?.isThereBranching) {
+          // presumes one single choice task per step
+          const [singleChoiceTask] = self.activeStepTasks.filter(task => task.type === 'single')
+          onAction(getRoot(self), (call) => {
+            if (call.path.endsWith(singleChoiceTask.annotation?.id) && call.name === 'update') {
+              let nextStepKey
+              const singleChoiceTaskAnswers = toJS(singleChoiceTask.answers)
+              const nextTaskKey = singleChoiceTaskAnswers[call.args[0]].next
+              self.steps.forEach(step => {
```

Comment:
This is only necessary when the single choice task is still the classifier 1.0 setup, that next is a task key rather than a step key. Maybe it can be like this:
```
              let nextStepKey
              const nextKey = singleChoiceTask.answers[call.args[0]].next
              if (nextKey.startsWith('T')) {
                // Backwards compatibility
                self.steps.forEach(step => {
                  if (step.taskKeys.includes(nextKey)) {
                    nextStepKey = step.stepKey
                  }
                })
              } else {
                nextStepKey = nextKey
              }
              self.active.setNext(nextStepKey)
            }
```